### PR TITLE
envoy: Update to 1.18.4

### PR DESCRIPTION
**Pull request by jrajahalme**
```release-note
Cilium Envoy integration is updated to Envoy release 1.18.4
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
gke-stable hit by #17204

File: pkg/envoy/server.go [hunk 1]
```diff
@@ -1011,7 +1011,6 @@ func createBootstrap(filePath string, nodeId, cluster string, xdsSock, egressClu
 			},
 		},
 		Admin: &envoy_config_bootstrap.Admin{
-			AccessLogPath: "/dev/null",
```
Comment (kkourt) [inline pkg/envoy/server.go:1014]: Was this intentional?

**Review by kkourt — APPROVED** [review]

File: pkg/envoy/server.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/envoy/server.go:1014]: Yes, sorry for not pointing this out in the commit message. `AccessLogPath` has been deprecated. Removing it eliminates a deprecation warning from Cilium agent logs.

**Review by aditighag — APPROVED** [review]

**Comment by jrajahalme** [comment]
net-next hit flake #17060

**Comment by jrajahalme** [comment]
EKS hit #16963

**Comment by jrajahalme** [comment]
Filed new issue #17260 for the Azure cluster creation failure.

**Comment by jrajahalme** [comment]
@joestringer None of the CI fails are related to Envoy. Should be safe to merge this bug fix as is.

**Pull request merged**